api/src/certification/scoring/domain/models/CertificationAssessmentScoreV3.js

```diff
@@ -64,41 +58,63 @@ class CertificationAssessmentScoreV3 {
   }
 }
 
-const _findIntervalIndex = (capacity, certificationScoringIntervals) =>
-  certificationScoringIntervals.findIndex(({ bounds }) => capacity <= bounds.max && capacity >= bounds.min);
-
-const _computeScore = ({
-  capacity,
-  maxReachableLevelOnCertificationDate,
-  certificationScoringIntervals,
-  intervalHeight,
-}) => {
-  let normalizedCapacity = capacity;
-  const minimumCapacity = certificationScoringIntervals[0].bounds.min;
-  const maximumCapacity = certificationScoringIntervals.at(-1).bounds.max;
-
-  if (normalizedCapacity < minimumCapacity) {
-    normalizedCapacity = minimumCapacity;
+const _findIntervalIndex = (capacity, certificationScoringIntervals) => {
+  if (capacity < certificationScoringIntervals[0].bounds.min) {
+    return 0;
   }
-  if (normalizedCapacity > maximumCapacity) {
-    normalizedCapacity = maximumCapacity;
+
+  for (const [index, { bounds }] of certificationScoringIntervals.entries()) {
+    if (bounds.max >= capacity) {
+      return index;
+    }
   }
 
-  const intervalIndex = _findIntervalIndex(normalizedCapacity, certificationScoringIntervals);
+  return certificationScoringIntervals.length - 1;
+};
+
+const _computeScore = ({ capacity, certificationScoringIntervals }) => {
+  const intervalIndex = _findIntervalIndex(capacity, certificationScoringIntervals);
+
+  const diff = capacity - certificationScoringIntervals[intervalIndex].bounds.max;
 
-  const intervalMaxValue = certificationScoringIntervals[intervalIndex].bounds.max;
   const intervalWidth =
     certificationScoringIntervals[intervalIndex].bounds.max - certificationScoringIntervals[intervalIndex].bounds.min;
 
-  // Formula is defined here : https://1024pix.atlassian.net/wiki/spaces/DD/pages/3835133953/Vulgarisation+score+2023#Le-score
-  const score = intervalHeight * (intervalIndex + 1 + (normalizedCapacity - intervalMaxValue) / intervalWidth);
+  return _compute({ certificationScoringIntervals, capacity, intervalIndex, diff, intervalWidth });
+};
+
+function _compute({ certificationScoringIntervals, capacity, intervalIndex, diff, intervalWidth }) {
+  const MAX_PIX_SCORE = 1024;
+  const numberOfIntervals = certificationScoringIntervals.length;
+  const SCORE_THRESHOLD = MAX_PIX_SCORE / numberOfIntervals;
+  const MAX_REACHABLE_LEVEL = 7;
+  const NUMBER_OF_COMPETENCES = 16;
+  const MIN_PIX_SCORE = 0;
+  const PIX_PER_LEVEL = 8;
+  const maximumReachableScore = MAX_REACHABLE_LEVEL * NUMBER_OF_COMPETENCES * PIX_PER_LEVEL - 1;
+
+  if (_isCapacityBelowMinimum(capacity, certificationScoringIntervals)) {
+    return MIN_PIX_SCORE;
+  } else if (_isCapacityAboveMaximum(capacity, certificationScoringIntervals)) {
+    return maximumReachableScore;
+  } else {
+    return _calculateScore({ intervalIndex, diff, intervalWidth, threshold: SCORE_THRESHOLD, maximumReachableScore });
+  }
```

Comment:
Neatpick pour alléger la lecture :

```suggestion
  if (_isCapacityBelowMinimum(capacity, certificationScoringIntervals)) {
    return MIN_PIX_SCORE;
  }
  
  if (_isCapacityAboveMaximum(capacity, certificationScoringIntervals)) {
    return maximumReachableScore;
  }
  
  return _calculateScore({ intervalIndex, diff, intervalWidth, threshold: SCORE_THRESHOLD, maximumReachableScore });
```